src/main/java/vectorwing/farmersdelight/common/block/MushroomColonyBlock.java

```diff
@@ -53,10 +59,11 @@ public class MushroomColonyBlock extends BushBlock implements BonemealableBlock
 			Block.box(1.0D, 0.0D, 1.0D, 15.0D, 14.0D, 15.0D),
 	};
 
-	public MushroomColonyBlock(Holder<Item> mushroomType, Properties properties) {
+	public MushroomColonyBlock(ItemLike mushroomType, Properties properties) {
```

Comment:
I don't think it's good to use `ItemLike` as the key, the `Item`, `Block`, `DefferedItem` and `DefferedBlock` all matches, but neither of them are equal to others by implementation.

Since you are using the `Item` to get value from the map, a modded colony which passed in a `DeferredItem` (or `DefferedBlock`) as the `mushroomType` parameter will simply fail when checked by the Rich Soil.

Please just revert this to use the `Holder<Item>` as the parameter type and key type, Holders are basically singleton and has proper `equals` and `hashCode` implementation on NeoForge, even on Fabric the Holders should still work totally fine as long as it's the `Holder.Reference` from registry.

Reply:
Alternatively just move the registration away from the constructor as requested above (the Apr 7 one), which would eliminate the need of laziness of the key.